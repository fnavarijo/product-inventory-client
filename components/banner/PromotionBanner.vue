<template>
  <div v-swiper="swiperOption">
    <div class="swiper-wrapper">
      <div
        v-for="(image, index) in images"
        :key="index"
        class="swiper-slide"
      >
        <img :src="image" alt="" class="w-full">
      </div>
    </div>
    <div class="swiper-pagination" />
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';

import { directive as swiper } from 'vue-awesome-swiper';

export default Vue.extend({
  directives: {
    swiper,
  },
  props: {
    images: {
      type: Array as PropType<Array<string>>,
      default: [],
    },
  },
  computed: {
    swiperOption () {
      return {
        swiperOption: {
          pagination: {
            el: '.swiper-pagination',
          },
          init: true,
          grabCursor: true,
        },
      };
    },
  },
});
</script>
