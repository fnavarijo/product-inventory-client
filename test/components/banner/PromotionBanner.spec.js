import { shallowMount } from '@vue/test-utils';

import PromotionBanner from '@/components/banner/PromotionBanner.vue';

describe('PromotionBanner component', () => {
  it('should receive images array as prop', () => {
    const wrapper = shallowMount(PromotionBanner, {
      propsData: {
        images: [
          'http://image.test.com/image/1',
          'http://image.test.com/image/2',
        ],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
