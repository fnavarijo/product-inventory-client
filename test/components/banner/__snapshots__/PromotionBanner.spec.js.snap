// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromotionBanner component should receive images array as prop 1`] = `
<div class="swiper-container swiper-container-initialized swiper-container-horizontal">
  <div class="swiper-wrapper" style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);">
    <div class="swiper-slide"><img src="http://image.test.com/image/1" alt="" class="w-full"></div>
    <div class="swiper-slide"><img src="http://image.test.com/image/2" alt="" class="w-full"></div>
  </div>
  <div class="swiper-pagination"></div><span class="swiper-notification" aria-live="assertive" aria-atomic="true"></span>
</div>
`;
